core/src/main/java/org/apache/iceberg/util/SnapshotUtil.java

```diff
@@ -397,6 +397,52 @@ public static Schema schemaFor(Table table, Long snapshotId, Long timestampMilli
     return table.schema();
   }
 
+  /**
+   * Return the schema of the snapshot at a given branch.
+   *
+   * <p>If branch does not exist, the table schema is returned because it will be the schema when
```

Comment:
The logic here for both finding schema and latest snapshot allows branch to be non existing. The reason I decided to do this way is because the core library will still allow auto-creation of branch, so it makes more sense to support that case for these util methods. We only block writing to non-existing branch through table identifier in Spark module, but we will support other cases like WAP branch that will leverage the core feature.

Reply:
Makes sense to me too.